scripts/serve/serve-static.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-console */
+const express = require('express');
+const path = require('path');
+const webpack = require('webpack');
+const MemoryFS = require('memory-fs');
+const mime = require('mime-types');
+
+const compile = (webpackConfig, vfs) => (
+  new Promise((resolve, reject) => {
+    const compiler = webpack(webpackConfig);
+    // setup a virtual file system to write webpack files to.
+    if (vfs) {
+      compiler.outputFileSystem = new MemoryFS();
+    }
+    console.log('[Terra-Toolkit:serve-static] Webpack compilation started');
+    compiler.run((err, stats) => {
+      if (err || stats.hasErrors()) {
+        console.log('[Terra-Toolkit:serve-static] Webpack compiled unsuccessfully');
+        reject(err || new Error(stats.toJson().errors));
+      } else {
+        console.log('[Terra-Toolkit:serve-static] Webpack compiled successfully');
+        resolve([webpackConfig.output.path, compiler.outputFileSystem]);
+      }
+    });
+  })
+);
+
+const generateSite = (site, config, vfs, production) => {
+  if (site) {
+    const sitePath = path.join(process.cwd(), site);
+    return Promise.resolve([sitePath, undefined]);
+  }
+
+  if (config) {
+    let webpackConfig = config;
+    if (typeof webpackConfig === 'function') {
+      webpackConfig = webpackConfig(undefined, { p: production });
+    }
+
+    return compile(webpackConfig, vfs);
+  }
+
+  return Promise.reject(new Error('No config provided.'));
+};
+
+const virtualApp = (site, index, fs) => {
+  const app = express();
+
+  // Setup a catch all route, we can't use 'static' because we need to use a virtual file system
```

Comment:
we can prob remove this comment now with having both static and virutual app methods

Reply:
This comment is still relevant about why we're not able to use serve static in this instance.